### PR TITLE
refactor: `first_name` and `last_name` move from `questionnaire` to `user`

### PR DESCRIPTION
**Pull request by peterkos**
Sub-implementation of #222, split into its own PR because it's a much bigger refactor than I anticipated. 

Current progress:
- [x] Models adjusted
- [x] Migration made (needs to be updated to a MOVE, not a delete)
- [x] Updated SOME tests and factories
- [x] Create account works
- [x] Add migration support for converting V1 database to V2 and back without losing any hackathon data
- [x] For migration, since V1 HM database doesn't have first/last names for admins, the migration will use the first half of the email (before `@`) as both the first/last name.
- [x] Stats page refactored
- [x] Fix all tests
- [x] Modified stats datatable to be statically setup (as all other datatables are)
- [x] Test lines no longer run (literally) 600+ lines long -_- (reindented inline requests)

And consolidating Chris's comment below:
- [x] Dietary Restrictions or Special Needs table in Stats is showing all users, not the preselected 10
- [x] Certain questionnaires raise a "undefined method `type' for nil:NilClass"
- [x] Users & Staff page not rendering either table properly (Admin table not loading and All Users not displaying names)
- [x] Viewing a bus list with passengers raises a "undefined method `full_name' for #Questionnaire:xxxxx"
Ajax error when attempting to open details on a particular school (Schools -> University at Buffalo)

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -34,7 +37,7 @@ def send_devise_notification(notification, *args)
 
   def queue_reminder_email
     return if reminder_sent_at
-    UserMailer.incomplete_reminder_email(id).deliver_later(wait: 1.day)
```
Comment (peterkos) [inline app/models/user.rb:37]: Whoops! Need to remove this.

Comment (cbaudouinjr) [reply app/models/user.rb:37]: Remove the whole line? That’s in place to remind users to finish applying if they haven’t within 24 hours of sign-up. What we could do is cancel the job if they’re promoted to a different role.

Comment (peterkos) [reply app/models/user.rb:37]: Whoops, sorry; I meant I needed to remove the change that commented that line out. 
Although that is an interesting idea... I'll add that to a permissions issue.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 class User < ApplicationRecord
-  audited only: [:email, :role, :is_active, :receive_weekly_report]
+  audited only: [:first_name, :last_name, :email, :role, :is_active, :receive_weekly_report]
+
+
```
[the remark this answers is not in the document]
Comment (peterkos) [reply app/models/user.rb:4]: GitHub/houndci-bot: Extra annoying bot detected.

**Comment by peterkos** [comment]
One quirk is that bus lists are no longer sorted by last name, but I figure we may refactor that soon anyways.

**Comment by cbaudouinjr** [comment]
Let's also merge in 2.0 during the testing process to make sure everything works with this major refactor.

**Comment by peterkos** [comment]
Merged 2.0 in once but yes, will merge again after I fix those UI bugs! (And the history crash)

**Comment by peterkos** [comment]
@cbaudouinjr For this one what was the console output for the error?
> Ajax error when attempting to open details on a particular school (Schools -> University at Buffalo)
Seems to work fine for me with an added person and without.

**Comment by peterkos** [comment]
>  Certain questionnaires raise a "undefined method `type' for nil:NilClass"

So, for this issue, the issue comes down that in `/app/views/manage/application/_model_history.html.haml`:
```ruby
- if audit.auditable.class.columns_hash[field].type == :text
```
The `audit.auditable.class.columns_hash[field]` is nil because the audit gem has no idea that the columns for first/last name have been migrated.

I've attempted a solution that just skips that check altogether and always shows the previous value. Seeeeems to work well? No crash. 
[2.0 branch]
Screen Shot 2020-06-30 at 12 43 15 AM

[suggested fix]
Screen Shot 2020-06-30 at 12 33 38 AM

I wasn't able to find any questionnaires that showed only `changed`. But for reference this is what it looks like if I force it to be triggered:

Screen Shot 2020-06-30 at 12 49 26 AM

Thoughts on risking a `changed from [null] to Baudouin` kind of error? I tried looking into the audit gem and couldn't find info on re-routing its info between migrations.

**Comment by peterkos** [comment]
> Ajax error when attempting to open details on a particular school (Schools -> University at Buffalo)
This is fixed in the previous commit, at the cost of this being the state of newly-added fields:
image

**Comment by peterkos** [comment]
@cbaudouinjr Could you double check the bus fix on your end? Just pushed a fix that should work

**Pull request merged**